3604/3/application/controllers/user_controller.rb

```diff
@@ -0,0 +1,49 @@
+class UserController < Sinatra::Base
+  use Rack::Flash
+
+  get '/sign_in' do
+    if session?
+      redirect '/main'
+    else
+      erb :sign_in
+    end
+  end
+
+  post '/sign_in' do
+    @user = User.find_by(email: params['email'])
+    if @user && @user.password == params[:password]
+      session_start!
+      session[:user_id] = @user.id
+      redirect '/main'
+    else
+      flash[:error] = I18n.t(:incorrect_sign_in)
+      redirect '/sign_in'
+    end
+  end
+
+  get '/sign_out' do
```

Comment:
this is semantic issue. you are destroying session, not fetching/reading. Choose DELETE or POST http method for this (preferable is delete)